oci/spec_opts.go

```diff
@@ -393,12 +396,77 @@ func WithImageConfigArgs(image Image, args []string) SpecOpts {
 			// even if there is no specified user in the image config
 			return WithAdditionalGIDs("root")(ctx, client, c, s)
 		} else if s.Windows != nil {
+			// imageExtended is a subset of the oci Image struct that changes
+			// the Config type to be imageConfigExtended in order to add the
+			// ability to deserialize `ArgsEscaped` which is not an OCI field,
+			// but is supported by Docker built images.
+			type imageExtended struct {
+				Config struct {
+					ArgsEscaped bool `json:"ArgsEscaped,omitempty"`
+				}
+			}
+			// Deserialize the extended image format for Windows.
+			var ociImageExtended imageExtended
+			if err := json.Unmarshal(imageConfigBytes, &ociImageExtended); err != nil {
+				return err
+			}
+			argsEscaped := ociImageExtended.Config.ArgsEscaped
+
 			s.Process.Env = replaceOrAppendEnvValues(config.Env, s.Process.Env)
-			cmd := config.Cmd
-			if len(args) > 0 {
-				cmd = args
+
+			if argsEscaped {
+				// For certain Windows images created via Docker they may
+				// contain `ArgsEscaped==true` if the ENTRYPOINT or CMD is in
+				// the `shell` form and contains spaces, path characters, etc.
+				// An example is the following:
+				//
+				// ```
+				// FROM mcr.microsoft.com/windows/servercore:ltsc2019
+				//
+				// SHELL ["powershell", "-Command"]
+				// ENTRYPOINT "C:\Path To\MyExe.exe" -arg1 "-arg2 value2"
+				// ```
+				//
+				// Will be encoded by Docker as:
+				// ```
+				// "Config": {
+				//   ...
+				//   "ArgsEscaped": true,
+				//   ...
+				//   "Entrypoint": [
+				//     "powershell.exe -Command \"C:\\My Data\\MyExe.exe\" -arg1 \"-arg2 value2\""
+				//   ],
+				//   ...
+				// }
+				// ```
+				//
+				// You can see that `Entrypoint` becomes a single element array
+				// with the entire argument list already escaped. To avoid a
+				// double escape problem Windows supports passing via OCI as a
+				// complete `CommandLine`.
+				if len(config.Entrypoint) == 1 {
+					// shell form ENTRYPOINT overrides shell form CMD so use it
+					// first.
+					s.Process.Args = nil
+					s.Process.CommandLine = config.Entrypoint[0]
+				} else if len(config.Cmd) == 1 {
+					// If no ENTRYPOINT then CMD should be used.
+					s.Process.Args = nil
+					s.Process.CommandLine = config.Cmd[0]
+				} else {
+					return errors.New("ImageConfig must contain either Entrypoint or Cmd when ArgsEscaped=true")
```

Comment:
I think we should do some extra sanity checks explicitly to make sure the image config appears "well formed". e.g. if `len(Entrypoint) > 1` and `len(Cmd) == 1`, return an error. It would also be nice to clarify in the error message specifically what was wrong (Entrypoint has too many elements, No Entrypoint/Cmd set, etc.).